model/task_lifecycle.go

```diff
@@ -41,12 +41,14 @@ func SetActiveState(taskId string, caller string, active bool) error {
 	if active {
 		// if the task is being activated and it doesn't override its dependencies
 		// activate the task's dependencies as well
+		tasksToActivate := []task.Task{}
 		if !t.OverrideDependencies {
-			for _, dep := range t.DependsOn {
-				if err = SetActiveState(dep.TaskId, caller, true); err != nil {
-					return errors.Wrapf(err, "error activating dependency for %v with id %v",
-						taskId, dep.TaskId)
-				}
+			deps, err := task.GetRecursiveDependenciesUp([]task.Task{*t}, nil)
```

Comment:
Will the result of GetRecursiveDependenciesUp include t? Or else should we add this at 44?

Reply:
it won't include t, but t is included later if we aren't resetting the task.